### PR TITLE
User Authentication

### PR DESCRIPTION
**Pull request by rennokki**
## Context

- PR: https://github.com/laravel/framework/pull/42531
- `signin()` method: https://pusher.com/docs/channels/using_channels/user-authentication/#sign-in

## Caveats

This version requires Pusher JS SDK 7.1+ (https://github.com/pusher/pusher-js/releases/tag/v7.1.0-beta)

This PR needs to be merged before:
- Documentation: https://github.com/laravel/docs/pull/7965

## Breaking Changes

None.

### CODE REVIEW

**Comment by taylorotwell** [comment]
So how would we ensure people have Pusher JS SDK 7.1+ installed?

**Comment by rennokki** [comment]
@taylorotwell

- The `signin()` prototype does not exist pre 7.1-beta. It's optional.
- This PR just adds the default endpoints that do not break already-existing apps. For apps that want to upgrade to JS SDK 7.1+, the defaults will point to the right user authentication endpoint. The endpoints are called only when `signin()` is used.

**Comment by jose123v** [comment]
Hi @rennokki, thanks for the great work

How to use this in soketi? is any change needed?

File: src/connector/connector.ts [hunk 1]
```diff
@@ -35,8 +39,11 @@ export abstract class Connector {
...
         this.options = Object.assign(this._defaultOptions, options);
 
-        if (this.csrfToken()) {
-            this.options.auth.headers['X-CSRF-TOKEN'] = this.csrfToken();
+        let token = this.csrfToken();
+
+        if (token) {
+            this.options.auth.headers['X-CSRF-TOKEN'] = token;
+            this.options.userAuthentication.headers['X-CSRF-TOKEN'] = token;
         }
```
Comment (jose123v) [inline src/connector/connector.ts:47]: @rennokki 
What about bearer token support?
https://github.com/laravel/echo/blob/f7e2373c31b0382bca8376d72f3258c29ea6a661/src/connector/connector.ts#L16-L18

```ts
if (token) {
    this.options.auth.headers['X-CSRF-TOKEN'] = token;
    this.options.userAuthentication.headers['X-CSRF-TOKEN'] = token;
} else if (this.options.bearerToken) {
    this.options.auth.headers['Authorization'] = 'Bearer '+ this.options.bearerToken;
    this.options.userAuthentication.headers['Authorization'] = 'Bearer '+ this.options.bearerToken;
}
```

Comment (jose123v) [reply src/connector/connector.ts:47]: @rennokki ping

**Pull request merged**